libvirt/uri/ssh.go

```diff
@@ -133,6 +133,11 @@ func (u *ConnectionURI) dialSSH() (net.Conn, error) {
 		username = sshu
 	}
 
+	hostname, err := sshcfg.Get(u.Hostname(), "HostName")
```

Comment:
`Get()` doesn't return an error when the key is not found, so this will result in the hostname ~never being read from the URI directly (i.e. it will _only_ work when there is an alias in `~/.ssh/config`).

And yes, the username thing immediately above is also broken.

Reply:
fyi, #1059 fixes all of these